tao_compiler/mlir/disc/IR/disc_shape_ops.td

```diff
@@ -73,4 +74,91 @@ def DISCShape_TieShapeOp: DISCShape_Op<"tie_shape", [NoSideEffect]> {
   let hasCanonicalizer = 1;
 }
 
+def DISCShape_SymbolicDimOp:
+      DISCShape_Op<"SymbolicDim", [IsolatedFromAbove, Symbol]> {
+  let summary = "representing a symbolic dimension size";
+  let description = [{
+    Representing a symbolic dimension size.
+  }];
+  let arguments = (ins
+    SymbolNameAttr:$sym_name,
+    DefaultValuedAttr<I64Attr, "-1">:$value,
+    DefaultValuedAttr<BoolAttr, "false">:$knownNonNegative,
+    DefaultValuedAttr<BoolAttr, "false">:$knownNegativeOne,
+    DefaultValuedAttr<BoolAttr, "false">:$knownNonSizeOne,
+    DefaultValuedAttr<BoolAttr, "false">:$knownNonSizeZero
+    // TODO(disc): add likely values
+  );
+  let extraClassDeclaration = [{
+    // Returns the dimension value this SymbolicDim op represents.
+    int64_t getDimSize();
+
+    // Sets the dimension value this SymbolicDim op represents to `val`
+    void setDimSize(int64_t val);
+
+    // Returns true if this SymbolicDim is not known at compile-time.
+    bool isDynamic();
+
+    // Returns true if we know that the dimension value this SymbolicDim op
+    // represents is size one.
+    bool knownSizeOne() { return getDimSize() == 1; }
+
+    // Sets `knownNonNegative` to the value of `flag`
+    void setKnownNonNegative(bool flag);
+
+    // Sets `knownNegativeOne` to the value of `flag`
+    void setKnownNegativeOne(bool flag);
+
+    // Sets `knownNonSizeOne` to the value of `flag`
+    void setKnownNonSizeOne(bool flag);
+
+    // Sets `knownNonSizeZero` to the value of `flag`
+    void setKnownNonSizeZero(bool flag);
+
+    // Try to merge two SymbolicDim ops.
```

Comment:
What does the 'merge' mean?

Reply:
If two SymbolicDim `a` and `b` are mapped to be equal, we will merge them in the union find set. Besides, we also merge the static known information of these two SymbolicDim ops. This merge function is just try to merge the static known information.

For example:
```
a is dynamic, b is const 2,  a merge b  -> a = b = 2
a is const 2, b is const 3, a merge b -> fail 
```